compose.py

```diff
@@ -0,0 +1,297 @@
+            libre2-dev \
+            libb64-dev \
+            pciutils && \
+    rm -rf /var/lib/apt/lists/*
+'''
+
+    # Add dependencies needed for python backend
+    if 'python' in FLAGS.backend:
+        df += '''
+# python3, python3-pip and some pip installs required for the python backend
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+         python3 \
+         python3-pip && \
+    pip3 install --upgrade pip && \
+    pip3 install --upgrade wheel setuptools && \
+    pip3 install --upgrade grpcio-tools grpcio-channelz numpy && \
+    rm -rf /var/lib/apt/lists/*
+'''
+    df += '''
+# Add entrypoint script
+WORKDIR /opt/tritonserver
+COPY --chown=1000:1000 --from=full /opt/tritonserver/nvidia_entrypoint.sh .
+ENTRYPOINT ["/opt/tritonserver/nvidia_entrypoint.sh"]
+'''
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+### Create container with docker build
+def build_docker_container(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument('--build-name',
+                        type=str,
+                        required=False,
+                        help='Build name. Default is "tritonserver".')
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        required=False,
+        help=
+        'Generated dockerfiles are placed here. Default to current directory.'
+    )
+    parser.add_argument(
+        '--version',
+        type=str,
+        required=False,
+        help=
+        'The Triton version. If not specified defaults to the value in the TRITON_VERSION file.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=False,
+        help=
+        'The Triton container version to build. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--upstream-container-version',
+        type=str,
+        required=False,
+        help=
+        'The upstream container version to use for the build. If not specified the upstream container version will be chosen automatically based on --version value.'
```

Comment:
"The version of the NGC Docker images to use as a base for this image. If not ..."